### PR TITLE
Fix named anchors

### LINKED ISSUE #260: Correct links with incorrect #named_anchor
## Issue 

While working on #259 it was discovered that our docs have a lot links that are valid, but have an invalid named anchor in the URL (the part after the `#`).

While not critical to fix, we should either decide to keep these up to date (and thus break the build if they're bad) or decide to do something less. Less might be checking them less frequently? I'm open to ideas!

Results were found with:

```bash
time muffet -c 1 -t 15  -e ".*npmjs\.com.*" -e ".*google\.com.*" -e ".*docker\.com.*" -e "http://localhost:5984.*" -e "https://demo.app.medicmobile.org"  http://localhost:1313
```

## Sample Result

http://YOUR_LOCAL_HUGO_INSTANCE/SPECIFIC/PAGE
 * id# BAD_NAME_HERE    http://TARGET_URL.example.com/WITH/MISSING/NAMED/ANCHOR

## Results - Special Fixes

http://localhost:1313/contribute/docs/style-guide/

 * ~~id#images-1 not found  https://www.markdownguide.org/basic-syntax/#images-1~~ NB - added with JS in browser, not detectable by `muffet`, is valid
 * ~~id#tables not found    https://www.markdownguide.org/extended-syntax/#tables~~  NB - added with JS in browser, not detectable by `muffet`, is valid

http://localhost:1313/apps/reference/contact-page/
 * ~~id#/guide/06_variable-replacement not found    https://angular-translate.github.io/docs/#/guide/06_variable-replacement~~  NB - deep link for angular-translate's SPA docs site, not detectable by `muffet`, is valid
 * ~~id#fields not found    http://localhost:1313/apps/reference/contact-page/#fields~~ fixed

http://localhost:1313/core/overview/translations/
 * ~~id#/guide/05_using-translate-directive not found       http://angular-translate.github.io/docs/#/guide/05_using-translate-directive~~  NB - deep link for angular-translate's SPA docs site, not detectable by `muffet`, is valid

http://localhost:1313/apps/guides/messaging/africas-talking/
 * ~~id#39;s Talking SMS Aggregator'&labels=enhancement&body=Page:+/apps/guides/messaging/africas-talking/
Version:+2020-06-05 21:05:09 -0400 -0400
--- not found   https://github.com/medic/cht-docs/issues/new?title=Improve+%27Africa&#39;s%20Talking%20SMS%20Aggregator%27&labels=enhancement&body=Page:+/apps/guides/messaging/africas-talking/%0AVersion:+2020-06-05%2021:05:09%20-0400%20-0400%0A---~~ fixed - NB - this will likely come back to haunt us as `&#39;` is the value of for URL encoded apostrophe which `muffet` interprets as an id.  Using a `’` instead of a `'` for now.

http://localhost:1313/apps/guides/messaging/message-loops/
 * ~~id#issuecomment-146254467 not found    https://github.com/medic/medic/issues/750#issuecomment-146254467~~ fixed NB - while the deep link to the exact fix was handy, you have to read the whole ticket for context any way, so a top level link is fine methinks

## Results - Normal Fixes

http://localhost:1313/apps/features/integrations/dhis2/

 * ~~id#get-apiv2exportdhis not found       https://github.com/medic/cht-core/tree/master/api#get-apiv2exportdhis~~ fixed

http://localhost:1313/apps/reference/forms/app/

 * ~~id#care-guides not found       http://localhost:1313/apps/reference/forms/contact/#care-guides~~ fixed
 * ~~id#icons not found     http://localhost:1313/apps/reference/forms/app/#icons~~ fixed
 * ~~id#localization not found      http://localhost:1313/apps/reference/forms/app/#localization~~ fixed

http://localhost:1313/apps/guides/database/rdbms-from-windows/
 * ~~id#tab-top not found   https://stackoverflow.com/questions/2224066/how-to-convert-ssh-keypairs-generated-using-puttygenwindows-into-key-pairs-use?answertab=votes#tab-top~~ fixed

http://localhost:1313/apps/guides/hosting/ec2-setup-guide/
 * ~~id#service-management-scripts not found        https://github.com/medic/medic-os#service-management-scripts~~ fixed

http://localhost:1313/apps/guides/hosting/self-hosting/
 * ~~id#service-management-scripts not found        https://github.com/medic/medic-os#service-management-scripts~~ fixed

http://localhost:1313/contribute/docs/workflow/
 * ~~id#installing-hugo not found   https://github.com/medic/cht-docs/blob/master/README.md#installing-hugo~~ fixed

http://localhost:1313/apps/concepts/prerequisites/
 * ~~id#installation not found      https://github.com/medic/medic-conf/blob/master/README.md#installation~~ fixed

http://localhost:1313/apps/guides/debugging/replicating-production-data-locally/
 * ~~id#what-it-does not found      https://github.com/pouchdb/add-cors-to-couchdb#what-it-does~~ fixed

http://localhost:1313/core/overview/data-flows-for-analytics/
 * ~~id#installation-steps-if-applicable not found  https://github.com/medic/medic-couch2pg#installation-steps-if-applicable~~ fixed

http://localhost:1313/core/guides/releasing/
 * ~~id#releasing not found https://github.com/medic/medic-conf/#releasing~~ fixed

http://localhost:1313/apps/reference/app-settings/outbound/
 * ~~id#post-apiv2records not found https://github.com/medic/medic/tree/master/api#post-apiv2records~~ fixed
 * ~~id#usage not found     https://github.com/mariocasciaro/object-path#usage~~ fixed

http://localhost:1313/apps/guides/forms/app-form-sms/
 * ~~id#compact-record-representation-(for-sms) not found   https://opendatakit.github.io/xforms-spec/#compact-record-representation-(for-sms)~~ fixed

http://localhost:1313/apps/guides/messaging/sms-states/
 * ~~id#content not found   https://github.com/medic/medic-gateway#content~~ fixed

http://localhost:1313/apps/guides/forms/multimedia/
 * ~~id#Browser_compatibility not found     https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats#Browser_compatibility~~ fixed

http://localhost:1313/apps/guides/data/csv-to-docs/
 * id#csv-file-name not found     https://github.com/medic/medic-conf#csv-file-name - __Link TBD__
 * ~~id#oauth2-client not found     https://github.com/googleapis/google-api-nodejs-client#oauth2-client~~ fixed

http://localhost:1313/apps/guides/debugging/obtaining-logs/
 * ~~id#medic-logs not found        https://github.com/medic/medic-conf#medic-logs~~ fixed

### PR DESCRIPTION
**Pull request by mrjones-plip**
Ready for review!

Fixes #260

### CODE REVIEW

**Comment by garethbowen** [comment]
@abbyad I think this PR highlights some issues with our CODEOWNERS set up.

Firstly, in the description @mrjones-plip has asked me not to review yet, but the system has automatically requested my review. Maybe we should ask that people don't create PRs until they're ready for review? Alternatively it's possible DRAFT PRs won't be automated in this way - we should explore and advise as to the best practice.

Secondly, this is a simple bug fix PR and yet the CODEOWNERS has pinged 3 people to review it. This has shifted our noise to signal ratio in the wrong direction. I think the problem is our ownership is too granular so it's quite likely a PR will notify multiple people. We should consider a less granular approach.

**Comment by mrjones-plip** [comment]
@abbyad  & @garethbowen  - oh yeah, I was curious about that! I saw that initially Max was added as a reviewer, but I didn't actually add him and, further, couldn't review him.  Anyway - happy to have this ticket be a guinea pig to iron out any kinks in the new system.  

Further - I like to be open about what I'm working on and create PRs very early for easy visibility with the diffs and files changes section of a PR.  Happy to not do this if we it creates too much noise (in old or new system)!

**Comment by garethbowen** [comment]
@mrjones-plip Have you seen GitHub's "draft PR" feature? I haven't used it much but I think it's exactly what you're after - opening a PR for transparency, but marking it as not ready to be merged. I'm hoping codeowners won't be assigned for review on PRs until they move out of draft stage...

**Comment by mrjones-plip** [comment]
@garethbowen  - Ah hah! Great great, I'll check out the [draft feature](https://github.blog/2019-02-14-introducing-draft-pull-requests/) - thanks! I see also you can convert an existing full PR [back to a draft](https://stackoverflow.com/questions/55070723/how-do-i-change-from-a-pr-to-a-draft-pr-at-github) in case I forget next time ;)

**Comment by abbyad** [comment]
I think the draft feature solves for early notifications, and we can see how to control who gets pinged so that internally we explicitly set the reviewers. If it is too noisy for you we can also remove you from the codeowners file while we work through these details.

**Comment by mrjones-plip** [comment]
Hey all! should I add another person to take the PR or is @michaelkohn , @MaxDiz  or @garethbowen  gonna grab this? It's not the most glamorous PR as it's a bunch of tedious validating of correct link fragments.  I am totally _selling_ this PR, aren't i!?!

**Comment by abbyad** [comment]
Switched the reviewer to Max alone -- thanks @MaxDiz!

File: content/en/apps/guides/messaging/africas-talking.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Africa's Talking SMS Aggregator
+title: Africa’s Talking SMS Aggregator
```
Comment (garethbowen) [inline content/en/apps/guides/messaging/africas-talking.md:2]: Are backticks better?

Comment (mrjones-plip) [reply content/en/apps/guides/messaging/africas-talking.md:2]: Well, my old boss would say that `’` is the proper apostrophe and `'` is, at least in the US, a [foot mark](https://practicaltypography.com/foot-and-inch-marks.html) ;)  Also, isn't that just a normal tick, not a backtick!?!  Too many rabbit holes, sorry ;)

In this case the `'` got rendered by `hugo` into `&#39` which `muffet` insisted was a URL fragment where as  `’` was something without the `#` character.  A very fragile change, I know. I should probably file a ticket against `muffet`, but after my bouts in the ring with FF and enketo, I was gonna sit this one out.

I called this out in the ticket with the list of links (see "Results - Special Fixes" on #260).

please unresolve if this doesn't resolve this!!

File: content/en/apps/features/integrations/dhis2.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ HRIOs access the **CHT App Management** tab and select the appropriate *DHIS2 Da
 
 ## DHIS2 apps
 
-The CHT also includes an [API](https://github.com/medic/cht-core/tree/master/api#get-apiv2exportdhis) that can be called from other applications that returns DHIS2 Data Sets. This means that you can build a [DHIS2 app](https://docs.dhis2.org/master/en/developer/html/apps_creating_apps.html) that pulls data from the CHT and imports it electronically into DHIS2. This would allow **Data Entry** to control the process directly from DHIS2 without having to access the CHT.
+The CHT also includes an [API](https://github.com/medic/cht-core/tree/master/api) that can be called from other applications that returns DHIS2 Data Sets. This means that you can build a [DHIS2 app](https://docs.dhis2.org/master/en/developer/html/apps_creating_apps.html) that pulls data from the CHT and imports it electronically into DHIS2. This would allow **Data Entry** to control the process directly from DHIS2 without having to access the CHT.
```
Comment (MaxDiz) [inline content/en/apps/features/integrations/dhis2.md:67]: this looks right but just takes me to the top of the cht-core/api page
https://github.com/medic/cht-core/tree/master/api#get-apiv2exportdhis

File: content/en/apps/guides/data/csv-to-docs.md [hunk 3]
```diff
@@ -295,7 +295,7 @@ To fetch the files from Google Drive run the `fetch-csvs-from-google-drive` acti
 
 #### Linking to Google Drive
 
-The file `csvs-on-google-drive.json` in your project's home directory will consist of a key value pair for each CSV file. The keys must be the filename where the CSV will be stored locally — see the CSV [file documentation](https://github.com/medic/medic-conf#csv-file-name) for the notation. The value of each key must be the ID of the corresponding file in Google Drive — the ID can be obtained from the URL eg `https://docs.google.com/spreadsheets/d/{FILE_ID}/edit`. 
+The file `csvs-on-google-drive.json` in your project's home directory will consist of a key value pair for each CSV file. The keys must be the filename where the CSV will be stored locally — see the CSV [file documentation](https://github.com/medic/medic-conf#user-content-csv-file-name) for the notation. The value of each key must be the ID of the corresponding file in Google Drive — the ID can be obtained from the URL eg `https://docs.google.com/spreadsheets/d/{FILE_ID}/edit`.
```
Comment (MaxDiz) [inline content/en/apps/guides/data/csv-to-docs.md:298]: this looks right but just takes me to the top of the medic-conf page
https://github.com/medic/medic-conf#csv-file-name

File: content/en/apps/guides/debugging/obtaining-logs.md [hunk 4]
```diff
@@ -22,6 +22,6 @@ The easiest way is to use `medic-logs`, a tool that comes with [`medic-conf`](ht
 medic-logs anInstance gardener
 ```
 
-See [medic-logs documentation](https://github.com/medic/medic-conf#medic-logs) for more details.
+See [medic-logs documentation](https://github.com/medic/medic-conf#user-content-fetching-logs) for more details.
```
Comment (MaxDiz) [inline content/en/apps/guides/debugging/obtaining-logs.md:25]: this looks right but just takes me to the top of the medic-conf page
https://github.com/medic/medic-conf#medic-logs

Comment (abbyad) [reply content/en/apps/guides/debugging/obtaining-logs.md:25]: Should probably be this link: https://github.com/medic/medic-conf#fetching-logs

Comment (MaxDiz) [inline content/en/apps/guides/debugging/obtaining-logs.md:25]: should probably be: https://github.com/medic/medic-conf#fetching-logs

**Comment by mrjones-plip** [comment]
Oh, shucks!! @MaxDiz  and @abbyad  - I should have called out that I took detailed notes in [the ticket itself](https://github.com/medic/cht-docs/issues/260) about each fix :(

for example, GH injects the anchors and hover links with JS after the DOM is settled (`onReady()` kinda stuffs).  These are, of course, invisible to `muffet` when it runs b/c it's just doing a simple `GET` of the raw HTML.  What is not invisible is the ones that are on the actual, initial HTML request. So https://github.com/medic/medic-conf#fetching-logs reports as broken, but you can just prepend `user-content-`  on there and it works for `muffet` and for actual humans: https://github.com/medic/medic-conf#user-content-fetching-logs

This is of course a royal pain for editors to do and semi-fragile.  I'm undecided to how to fix going foward.

For now, following Gareth's MVP approach, we're launching the [link checker](https://github.com/medic/cht-docs/issues/259) with the fragment checking disabled.  None the less, we had a lot of broken content that this PR fixes, so a net win either way we proceed.

File: content/en/apps/guides/data/csv-to-docs.md (hunk 3 above)
Comment (mrjones-plip) [reply content/en/apps/guides/data/csv-to-docs.md:298]: oh - so it does! I'm not actually sure if there's a logical link on this page for "file documentation". I'd guess ["managing data and images"](https://github.com/medic/medic-conf#managing-data-and-images) maybe?

File: content/en/apps/guides/debugging/obtaining-logs.md (hunk 4 above)
Comment (mrjones-plip) [reply content/en/apps/guides/debugging/obtaining-logs.md:25]: yes, both work! see my '[oh shucks](https://github.com/medic/cht-docs/pull/261#issuecomment-671624264)' comment about this PR wide change I made for fragment.

Comment (MaxDiz) [reply content/en/apps/guides/debugging/obtaining-logs.md:25]: cool, leaving as is then

File: content/en/apps/guides/data/csv-to-docs.md (hunk 3 above)
Comment (MaxDiz) [reply content/en/apps/guides/data/csv-to-docs.md:298]: or maybe "[csv to docs](https://docs.communityhealthtoolkit.org/apps/guides/data/csv-to-docs/)" ?
@abbyad do you know?

Comment (abbyad) [reply content/en/apps/guides/data/csv-to-docs.md:298]: @MaxDiz, that's the file in question, right? I think that's a good start, but I would go with the [#creating-csv-files-for-contacts-reports](https://docs.communityhealthtoolkit.org/apps/guides/data/csv-to-docs/#creating-csv-files-for-contacts-reports) anchor. In general, if we are linking to documentation content in our own github repos it probably means that we need to port content over to the doc site. I think in this case we already did!

Comment (mrjones-plip) [reply content/en/apps/guides/data/csv-to-docs.md]: ack!  So sorry I missed this response.  I committed the suggested link -  thank you!

**Comment by mrjones-plip** [comment]
FYI - I fixed a stack exchange link that was using the short link (`https://stackoverflow.com/a/2224204`) to use the full link (`https://stackoverflow.com/questions/2224066/how-to-convert-ssh-keypairs-generated-using-puttygen-windows-into-key-pairs-us/2224204#2224204`) as `muffet` was unhappy using the short one for some reason:

```
All Branches - Run Muffet link checker
Run ./.github/scripts/muffet.sh

http://localhost:1313/apps/guides/database/rdbms-from-windows/
	400	https://stackoverflow.com/a/2224204
##[error]Process completed with exit code 1.
```

see eefa043

**Comment by mrjones-plip** [comment]
sigh - sorry if @garethbowen  just got a re-review request, meant to send it only to @MaxDiz .  I'll get git right one of these days ;)

**Comment by mrjones-plip** [comment]
Oh - actually, I forgot @MaxDiz  is on leave - @garethbowen  would you mind sitting in for him to close on the review on this PR?

**Review by garethbowen — APPROVED** [review]
LGTM. Merge when ready.

**Pull request merged**